kubernetes/metadata/metadata.go

```diff
@@ -134,19 +133,23 @@ func GetKubernetesClusterIdentifier(cfg *config.C, client k8sclient.Interface) (
 	if err == nil {
 		return clusterInfo, nil
 	}
-	// try with kubeadm-config configmap
-	clusterInfo, err = getClusterInfoFromKubeadmConfigMap(client)
+
+	// try with kubeadm-config configmap only if config_kubeAdm == true
+	clusterInfo, err = getClusterInfoFromKubeadmConfigMap(client, c.KubeAdm)
 	if err == nil {
 		return clusterInfo, nil
 	}
 	return ClusterInfo{}, fmt.Errorf("unable to retrieve cluster identifiers")
 }
 
-func getClusterInfoFromKubeadmConfigMap(client k8sclient.Interface) (ClusterInfo, error) {
+func getClusterInfoFromKubeadmConfigMap(client k8sclient.Interface, kubeadm bool) (ClusterInfo, error) {
 	clusterInfo := ClusterInfo{}
 	if client == nil {
 		return clusterInfo, fmt.Errorf("unable to get cluster identifiers from kubeadm-config")
 	}
+	if kubeadm {
+		return clusterInfo, fmt.Errorf("unable to get cluster identifiers from kubeadm-config because conf_kubeadm access has been disabled")
+	}
```

Comment:
One question, will this be reported as an error in the logs? It seems to me if the user disabled that, it should not be reported as an error.

Reply:
Good point. I will change it to Debug

Reply:
Reverted to nil